src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -240,7 +241,12 @@ pub(super) struct Node {
     pub children: SmallVec<[UniIndex; 4]>,
     /// Either `Reserved` or `Frozen`, the permission this tag will be lazily initialized
     /// to on the first access.
+    /// Can also be `Active` for the root node, in which case we consider every node
+    /// to be already initialized (see `is_default_initialized`).
     default_initial_perm: Permission,
+    /// Whether this node is always considered to be initialized. This is `true` for the
+    /// root node, which is also the only by-default `Active` node.
+    is_default_initialized: bool,
```

Comment:
Why do you need this?

Reply:
Turns out it is not necessary. Instead, we've decided to implement a different approach (see comment below)